rust/connlib/tunnel/src/peer.rs

```diff
@@ -260,8 +260,24 @@ impl ClientOnGateway {
     }
 
     pub(crate) fn expire_resources(&mut self, now: DateTime<Utc>) {
-        self.resources.retain(|_, r| r.is_allowed(&now));
-        self.recalculate_filters();
+        let cid = self.id;
+        let mut any_expired = false;
+
+        self.resources.retain(|rid, r| {
+            let is_allowed = r.is_allowed(&now);
+
+            if !is_allowed {
+                tracing::info!(%cid, %rid, "Access to resource expired");
+            }
+
+            any_expired |= is_allowed;
+
+            is_allowed
+        });
+
+        if any_expired {
+            self.recalculate_filters();
+        }
```

Comment:
This change was to print the expire log and re-calculate the filters less often? Otherwise it should be equivalent right?

Reply:
> This change was to print the expire log and re-calculate the filters less often? Otherwise it should be equivalent right?

Yeah, also a gateway may have lots of resources and they usually don't expire so I didn't want to run the filter recalculation every second or whatever it is being called at.